text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,703 @@
+
+## Data Model
+
+This section describes various protobuf messages that are used to represent profiles data.
+
+### Relationships Diagram
+
+The following diagram shows the relationships between the messages. Relationships between messages are represented by either embedding one message in another (red arrows), or by referencing a message by index in a lookup table (blue arrows). More on that in [Relationships between messages](#relationships-between-messages) section below.
+
+In addition to that, relationship between `stacktraces`, `attribute_sets` and `links` is implicit and based on the order of references to these objects in corresponding reference lists within a `ProfileType` message. The relationship between these messages creates an ephemeral structure called "Sample". More on that in [Arrays of Integers vs Arrays of Structures](#arrays-of-integers-vs-arrays-of-structures) section below.
+
+![diagram of data relationships](./images/otep0237/profiles-data-model.png)
+
+### Relationships With Other Signals
+
+There are two types of relationships between profiles and other signals:
+
+* from other signals to profiles (e.g from log records, exemplars or trace spans)
+* from profiles to other signals
+
+#### From profiles to other signals
+
+[Link](#message-link) is a message that is used to represent connections between profile [Samples](#message-sample) and trace spans. It uses `trace_id` and `span_id` as identifiers.
+
+For other signals, such as logs or metrics, because other signals use the same way of linking between such signals and traces (`trace_id` and `span_id`), it is possible to correlate profiles with other signals using this same information.
+
+#### From other signals to profiles
+
+Other signals can use `profile_id` to reference a profile. For example, a log record can reference a profile that was collected at the time when the log record was generated by using `profile_id` as one of the attributes. This allows to correlate logs with profiles.
+
+Additionally, `trace_id`, `span_id` can be used to reference specific [Samples](#message-sample) in a Profile, since [Samples](#message-sample) are linked to traces with these same identifiers using [Links](#message-link).
+
+The exact details of such linking are out of scope for this OTEP. It is expected that other OTEPs will define how such linking is done.
+
+### Message Descriptions
+
+These are detailed descriptions of protobuf messages that are used to represent profiling data.
+
+<!-- messages -->
+#### Message `ProfilesData`
```

Comment:
Is there a reason why the .proto file is not included directly in this OTEP? I find this markdown harder to read and understand. Is the intent here to describe a data model without tying it to the Protobuf format?

Reply:
Updated the OTEP with the proto.